lib/src/iterable_extensions.dart

```diff
@@ -0,0 +1,38 @@
+import 'objects.dart';
+
+/// Extends [Iterable]s with some common helper properties and functions
+///
+/// `since 0.0.1`
+extension IterableExtensions<E> on Iterable<E> {
+  /// Returns the first element from the iterable, or null if the iterable is
+  /// empty
+  ///
+  /// `since 0.0.1`
+  E? get firstOrNull {
+    var it = iterator;
+    return it.moveNext() ? it.current : null;
```

Comment:
This should be extracted to a pure function in `collections.dart`. Then
```suggestion
    return coll.firstOrNull<E>(this);
```
```dart
E? firstOrNull<E>(Iterable<E> iterable) {
  var it = iterable.iterator;
  return it.moveNext() ? it.current : null;
}
```